src/dataset.jl

```diff
@@ -87,6 +106,30 @@ end
 Returns a mapping of constraint patterns to specific dimension paths.
 The returned dictionary has keys of type [`Pattern`](@ref) and the values are sets of
 `Tuple{Vararg{Symbol}}`.
+
+# Example
+```jldoctest
+julia> using AxisKeys; using AxisSets: Dataset, constraintmap;
+
+julia> ds = Dataset(
+           :val1 => KeyedArray(rand(4, 3, 2); time=1:4, loc=-1:-1:-3, obj=[:a, :b]),
+           :val2 => KeyedArray(rand(4, 3, 2) .+ 1.0; time=1:4, loc=-1:-1:-3, obj=[:a, :b]),
+       );
+
+julia> cmap = constraintmap(ds);
+
+julia> keys(cmap)
+Base.KeySet for a OrderedCollections.LittleDict{AxisSets.Pattern,Set{Tuple{Vararg{Symbol,N} where N}},Array{AxisSets.Pattern,1},Array{Set{Tuple{Vararg{Symbol,N} where N}},1}} with 3 entries. Keys:
```

Comment:
Yeah, actually, I think calling `collect` on the `constraintmap` output directly might make this work. Documenter was getting annoyed about inconsistent line lengths locally vs on CI.

Reply:
Yeah, actually, `collect` could probably make `collect(constraintmap(ds))` work properly in documneter. I was getting inconsistent newlines because the types were sufficiently complex.